### PR TITLE
Clarify `is_polygon_clockwise()` coordinate system

### LINKED ISSUE #49716: `Geometry.is_polygon_clockwise` seems to return the opposite of expected results
### Godot version

v3.3.2.stable.official

### System information

Windows 10, GLES3,  GeForce GTX 1070/PCIe/SSE2

### Issue description

The results of `Geometry.is_polygon_clockwise` seem consistent with a cartesian coordinate system where the Y axis grows upwards. However, I was under the impression that 2D graphics in Godot used a coordinate system where the Y axis grows downward, and since `Geometry.is_polygon_clockwise` is an inherently 2D function, the results given seem to be the opposite of what I would have expected.

### Steps to reproduce

1. Add the following line to any node script and observe that the output is `True`, which is not consistent with the 2D coordinate system that Godot uses with the Y axis growing downward.

```gdscript
print(str(Geometry.is_polygon_clockwise(PoolVector2Array([Vector2(-1, -1), Vector2(-1, 1), Vector2(1, 1), Vector2(1, -1)]))))
```

### Minimal reproduction project

_No response_

### LINKED ISSUE #92146: Geometry2D.merge_polygons will always return polygons with counter-clock-wise
### Tested versions

Godot v4.3.dev6.mono

### System information

Windows 10.0.22631 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3050 Laptop GPU (NVIDIA; 31.0.15.2698) - AMD Ryzen 7 5800U with Radeon Graphics (16 Threads)

### Issue description

This code merges two unit (clock-wise / counter-clock-wise) sqaure polygons, but returns the same result. 
The result polygons are counter-clock-wise. 

According to the document, it should returns a cw polygon if it is a boundary and a ccw one if a hole. 

```gdscript
extends Node

func _ready() -> void:
	var poly1 = [Vector2(0, 0), Vector2(0, 1), Vector2(1, 1), Vector2(1, 0)]
	var poly2 = [Vector2(1, 0), Vector2(1, 1), Vector2(2, 1), Vector2(2, 0)]
	print(Geometry2D.is_polygon_clockwise(poly1))
	print(Geometry2D.is_polygon_clockwise(poly2))
	var merged = Geometry2D.merge_polygons(poly1, poly2)
	poly1.reverse()
	poly2.reverse()
	print(poly1)
	print(poly2)
	var reversed_merged = Geometry2D.merge_polygons(poly1, poly2)
	print(merged)
	print(reversed_merged)
	for poly in merged:
		print(Geometry2D.is_polygon_clockwise(poly))
	for poly in reversed_merged:
		print(Geometry2D.is_polygon_clockwise(poly))
	pass

```

logs:
```
true
true
[(1, 0), (1, 1), (0, 1), (0, 0)]
[(2, 0), (2, 1), (1, 1), (1, 0)]
[[(2, 0), (2, 1), (0, 1), (0, 0)]]
[[(2, 0), (2, 1), (0, 1), (0, 0)]]
false
false
```

### Steps to reproduce

Create a node, copy and apply the script, and run. 

### Minimal reproduction project (MRP)

N/A

### PR DESCRIPTION
**Pull request by lawnjelly**
The function assumes a normal Cartesian coordinate system.

Fixes #49716

## Notes
* Alternative would be to add a parameter to do the flip as in #89635, but that may be controversial, and is taking a while to be decided on, so this doc change will do the job either as the final solution, or until we decide on another solution.

### CODE REVIEW

File: doc/classes/Geometry2D.xml [hunk 1]
```diff
@@ -116,6 +116,8 @@
 			<param index="0" name="polygon" type="PackedVector2Array" />
 			<description>
 				Returns [code]true[/code] if [param polygon]'s vertices are ordered in clockwise order, otherwise returns [code]false[/code].
+				[b]Note:[/b] Assumes a Cartesian coordinate system where [code]+x[/code] is right and [code]+y[/code] is up.
+				If using screen coordinates ([code]+y[/code] is down), the result will need to be flipped (i.e. a [code]true[/code] result will indicate counter-clockwise).
```
Comment (AThousandShips) [inline doc/classes/Geometry2D.xml:120]: Should be one paragraph IMO

**Review by AThousandShips — APPROVED** [review]

**Comment by AThousandShips** [comment]
Will likely be needed for older versions regardless of adding a handedness option down the line (and likely won't get into 4.3 anyway)

File: core/math/geometry_2d.h [hunk 2]
```diff
@@ -350,6 +350,8 @@ class Geometry2D {
 		return triangles;
 	}
 
+	// Assumes cartesian coordinate system (e.g. +x to the right, +y up).
+	// If using screen coordinates (+y down) the result will need to be flipped.
```
Comment (akien-mga) [inline core/math/geometry_2d.h:354]: The description in the docs is more correct IMO. The screen coordinate system is also Cartesian AFAIK, i.e. defined by two orthogonal axes that cross at the origin.

From Wikipedia:

> In mathematics, physics, and engineering, the first axis is **usually** defined or depicted as horizontal and oriented to the right, and the second axis is vertical and oriented upwards. **(However, in some [computer graphics](https://en.wikipedia.org/wiki/Computer_graphics) contexts, the ordinate axis may be oriented downwards.)** The origin is often labeled O, and the two coordinates are often denoted by the letters X and Y, or x and y. The axes may then be referred to as the X-axis and Y-axis. The choices of letters come from the original convention, which is to use the latter part of the alphabet to indicate unknown values. The first part of the alphabet was used to designate known values.

(Emphases mine.)

Well long comment to say it shouldn't say "e.g. +x to the right, +y up" - it's not an example, it's the requirement.

Comment (lawnjelly) [reply core/math/geometry_2d.h]: Yeah I was admittedly a bit more careful for the docs version as more people will read it. :grin:

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**